stdlib/public/core/Array.swift

```diff
@@ -1844,9 +1878,15 @@ extension Array {
     let insertCount = newElements.count
     let growth = insertCount - eraseCount
 
-    _reserveCapacityImpl(minimumCapacity: self.count + growth,
-                         growForAppend: true)
-    _buffer.replaceSubrange(subrange, with: insertCount, elementsOf: newElements)
+    let punchedHole = _reserveCapacityImplForReplaceSubrange(
+      minimumCapacity: self.count + growth,
+      growForAppend: growth > 0,
```

Comment:
The change to `growth > 0` instead of `true` is interesting to note. It should mean we use a bit less memory, but that can have odd perf consequences.

Reply:
I decided this was actually not right, so I'm changing it back